File: android/src/main/java/cloud/blinklabs/yoroi/cslmobilebridge/CslMobileBridgeModule.java
package cloud.blinklabs.yoroi.cslmobilebridge;

import androidx.annotation.Nullable;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.turbomodule.core.interfaces.TurboModule;
import com.facebook.react.bridge.ReactContextBaseJavaModule;

public class CslMobileBridgeModule extends ReactContextBaseJavaModule implements TurboModule {
  public static final String NAME = "CslMobileBridge";

  public CslMobileBridgeModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return NAME;
  }
}

File: android/src/main/java/com/emurgo/cslmobilebridge/CslMobileBridgeModule.kt
package com.emurgo.cslmobilebridge

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = CslMobileBridgeModule.NAME)
class CslMobileBridgeModule(reactContext: ReactApplicationContext) :
  NativeCslMobileBridgeSpec(reactContext) {

  override fun getName(): String {
    return NAME
  }

  override fun initialize() {
    super.initialize()
    install()
  }

  private external fun install()

  companion object {
    const val NAME = "CslMobileBridge"

    init {
      System.loadLibrary("react_native_csl_mobile_bridge")
    }
  }
}
